File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
# 앞에 글자가 가장 큰 수를 처음에 넣기
def solution(numbers):
    numbers = list(map(str, numbers))
    # 1000까지 해당하니까 4자리수까지 포함
    numbers.sort(key=lambda x: x * 4, reverse=True)
    answer = str(int(''.join(numbers)))
    return answer

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.js
function solution(numbers) {

  const strNumber = numbers.map(String)
  // 정렬
  strNumber.sort((a, b) => (b + a) - (a + b))
  
  // 문자열로 바꾸어 출력
  result = strNumber.join('')

  // 결과값이 0이면 0 반환
  return result[0] === '0' ? '0' : result
}